huaweicloud/config/config.go

```diff
@@ -371,6 +401,59 @@ func genClients(c *Config, pao, dao golangsdk.AuthOptionsProvider) error {
 	return err
 }
 
+func getAuthConfigByMeta(c *Config) error {
+	req, err := http.NewRequest("GET", securityKeyURL, nil)
+	req.Header.Set("Content-Type", "application/json")
+	if err != nil {
+		return fmt.Errorf("Error building metadata API request: %s", err.Error())
+	}
+	httpClient := &http.Client{}
```

Comment:
should we add the user-agent and log the request?

Reply:
done